packages/opencensus-core/src/metrics/gauges/gauge.ts

```diff
@@ -0,0 +1,170 @@
+
+  /**
+   * Removes the TimeSeries from the gauge metric, if it is present. i.e.
+   * references to previous Point objects are invalid (not part of the
+   * metric).
+   *
+   * @param {LabelValue[]} labelValues The list of label values.
+   */
+  removeTimeSeries(labelValues: LabelValue[]): void {
+    validateNotNull(labelValues, 'labelValues');
+    this.registeredPoints.delete(hashLabelValues(labelValues));
+  }
+
+  /**
+   * Removes all TimeSeries from the gauge metric. i.e. references to all
+   * previous Point objects are invalid (not part of the metric).
+   */
+  clear(): void {
+    this.registeredPoints.clear();
+  }
+
+  /**
+   * Registers a TimeSeries and returns a Point if the specified
+   * labelValues is not already associated with this gauge, else returns an
+   * existing Point.
+   *
+   * @param {LabelValue[]} labelValues The list of the label values.
+   * @returns {types.Point} The value of single gauge.
+   */
+  private registerTimeSeries(labelValues: LabelValue[]): types.Point {
+    const hash = hashLabelValues(labelValues);
+    // return if the specified labelValues is already associated with the point.
+    if (this.registeredPoints.has(hash)) {
+      return this.registeredPoints.get(hash);
+    }
+    if (this.labelKeysLength !== labelValues.length) {
+      throw new Error('Label Keys and Label Values don\'t have same size');
+    }
+
+    const point = new PointEntry(labelValues);
+    this.registeredPoints.set(hash, point);
+    return point;
+  }
+
+  /**
+   * Provides a Metric with one or more TimeSeries.
+   *
+   * @returns {Metric} The Metric.
+   */
+  getMetric(): Metric {
+    if (this.registeredPoints.size === 0) {
+      return null;
+    }
+    const hrtimeLocal = process.hrtime();
+    return {
+      descriptor: this.metricDescriptor,
+      timeseries: Array.from(
+          this.registeredPoints,
+          ([_, point]) => point.getTimeSeries(
+              {seconds: hrtimeLocal[0], nanos: hrtimeLocal[1]}))
+    };
+  }
+}
+
+/**
+ * The value of a single point in the Gauge.TimeSeries.
+ */
+export class PointEntry implements types.Point {
+  private value = 0;
+
+  constructor(private readonly labelValues: LabelValue[]) {
+    this.labelValues = labelValues;
+  }
+
+  /**
+   * Adds the given value to the current value. The values can be negative.
+   *
+   * @param {number} amt The value to add.
+   */
+  add(amt: number): void {
+    this.value = this.value + amt;
+  }
+
+  /**
+   * Sets the given value.
+   *
+   * @param {number} val The new value.
+   */
+  set(val: number): void {
+    this.value = val;
+  }
+
+  /**
+   * Returns the TimeSeries with one or more Point.
+   *
+   * @param {Timestamp} timestamp The time at which the gauge is recorded.
+   * @returns {TimeSeries} The TimeSeries.
+   */
+  getTimeSeries(timestamp: Timestamp): TimeSeries {
+    return {
+      labelValues: this.labelValues,
+      points: [{value: this.value, timestamp}],
+      startTimestamp: null
```

Comment:
Why its `null`

Reply:
```start_timestamp```: Must be present for cumulative metrics only. https://github.com/census-instrumentation/opencensus-proto/blob/master/src/opencensus/proto/metrics/v1/metrics.proto#L133-L137

Reply:
Is it optional param? if yes - please change it in type

Reply:
Done